modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/application/authz/ApplicationAuthzTenantTestCase.java

```diff
@@ -0,0 +1,192 @@
+        entitlementPolicyClient = new EntitlementPolicyServiceClient(backendURL, sessionCookie);
+
+        httpClientAzUser = new DefaultHttpClient();
+        httpClientNonAzUser = new DefaultHttpClient();
+
+        createRole(AZ_TEST_TENANT_ROLE);
+        createUser(AZ_TEST_TENANT_USER, AZ_TEST_TENANT_USER_PW, new String[]{AZ_TEST_TENANT_ROLE});
+        createUser(NON_AZ_TEST_TENANT_USER, NON_AZ_TEST_TENANT_USER_PW, new String[0]);
+        createApplication(APPLICATION_NAME);
+        createSAMLApp(APPLICATION_NAME, true, false, false);
+        setupXACMLPolicy(POLICY_ID, POLICY);
+
+        //Starting tomcat
+        log.info("Starting Tomcat");
+        tomcatServer = Utils.getTomcat(getClass());
+
+        URL resourceUrl = getClass()
+                .getResource(File.separator + "samples" + File.separator + "travelocity.com-saml-tenantwithoutsigning.war");
+        Utils.startTomcat(tomcatServer, "/" + APPLICATION_NAME, resourceUrl.getPath());
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        deleteUser(AZ_TEST_TENANT_USER);
+        deleteUser(NON_AZ_TEST_TENANT_USER);
+        deleteRole(AZ_TEST_TENANT_ROLE);
+        deleteApplication(APPLICATION_NAME);
+        entitlementPolicyClient.removePolicy(POLICY_ID);
+
+        ssoConfigServiceClient = null;
+        applicationManagementServiceClient = null;
+        remoteUSMServiceClient = null;
+        httpClientAzUser = null;
+        //Stopping tomcat
+        tomcatServer.stop();
+        tomcatServer.destroy();
+        Thread.sleep(10000);
+    }
+
+    @Test(alwaysRun = true, description = "Test authorized tenant user login by evaluating the policy", groups = "wso2.is")
+    public void testAuthorizedSAMLSSOLogin() {
+
+        try {
+
+            HttpResponse response;
+
+            response =
+                    Utils.sendGetRequest(String.format(SAML_SSO_LOGIN_URL, APPLICATION_NAME, HTTP_REDIRECT), USER_AGENT,
+                            httpClientAzUser);
+            String sessionKey = Utils.extractDataFromResponse(response, CommonConstants.SESSION_DATA_KEY, 1);
+            response = Utils.sendPOSTMessage(sessionKey, COMMON_AUTH_URL, USER_AGENT, ACS_URL, APPLICATION_NAME,
+                    AZ_TEST_TENANT_USER + WSO2_DOMAIN, AZ_TEST_TENANT_USER_PW, httpClientAzUser);
+            EntityUtils.consume(response.getEntity());
+
+            response = Utils.sendRedirectRequest(response, USER_AGENT, ACS_URL, APPLICATION_NAME,
+                    httpClientAzUser);
+            String samlResponse = Utils.extractDataFromResponse(response, CommonConstants.SAML_RESPONSE_PARAM, 5);
+
+            response = sendSAMLMessage(String.format(ACS_URL, APPLICATION_NAME), CommonConstants
+                    .SAML_RESPONSE_PARAM, samlResponse);
+            String resultPage = extractDataFromResponse(response);
+
+            Assert.assertTrue(resultPage.contains("You are logged in as " + AZ_TEST_TENANT_USER),
```

Comment:
The assertion message should be what we expect. 
e.g "SAML SSO Login should be success and page should have a message \" You are logged in as "+ AZ_TEST_TENANT_USER